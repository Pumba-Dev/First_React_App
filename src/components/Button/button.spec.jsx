import { render, screen, fireEvent} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "./"

describe ('<button />', () => {
    
    it('should render the button with text', () => {
        render(<Button text="text test" />);
        const $button = screen.getByRole('button', { name: /text test/i });
        expect($button).toBeInTheDocument();
    });

    it('should call function on button click', () => {
        const mockFunction = jest.fn();
        render(<Button text="text test" onClick={mockFunction} ></Button>);
        const $button = screen.getByRole('button', { name: /text test/i });
        userEvent.click($button);
        expect(mockFunction).toHaveBeenCalledTimes(1);
    });

    it('should be desable when disable is true', () => {
        const mockFunction = jest.fn();
        render(<Button text="text test" onClick={mockFunction} disabled={true} />);
        const $button = screen.getByRole('button', { name: /text test/i });
        expect($button).toBeDisabled();
    }); 

    it('should snapshot match', () => {
        const mockFunction = jest.fn();
        const {container} = render(<Button text="text test" onClick={mockFunction} disabled={true} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});