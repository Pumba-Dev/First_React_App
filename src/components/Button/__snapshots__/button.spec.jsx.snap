// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<button /> should snapshot match 1`] = `
<button
  class="button"
  disabled=""
>
  text test
</button>
`;
